### PR TITLE
Bulk

### PR DESCRIPTION
**Pull request by vwidjaya**
- refactor for go modules
- refactor to use bulk inserts to es
- create only 1 es client instance for each consumer worker; previously a new es client is made on every Store()
- implement ticker that logs estimated TPS
- allow optional bulk insert vs single insert
- allow configs for bulk insert trigger
- fix go mod and refactor test
- make tps benchmark optional

### CODE REVIEW

File: flow/elastic.go [hunk 1]
```diff
@@ -19,10 +21,29 @@ type Elastic interface {
...
 }
 
-func NewElastic(retrierFunc *ElasticRetrier, urls ...string) (client elasticClient, err error) {
+func NewEsConfig(indexMethod string, bulkSize int, flushMs time.Duration, printTPS bool) esConfig {
+
+	return esConfig{
+		indexMethod:  indexMethod,
+		bulkSize:     bulkSize,
+		flushMs:      flushMs,
+		printTPS:	  printTPS,
```
Comment (giosakti) [inline flow/elastic.go]: have you run `go fmt` for this file?

File: flow/elastic.go [hunk 2]
```diff
@@ -31,9 +52,42 @@ func NewElastic(retrierFunc *ElasticRetrier, urls ...string) (client elasticClie
 		elastic.SetRetrier(retrierFunc),
 	)
 
-	return elasticClient{
+	p, err := c.BulkProcessor().
+		BulkActions(esConfig.bulkSize).
+		FlushInterval(esConfig.flushMs * time.Millisecond).
```
Comment (giosakti) [inline flow/elastic.go]: so now do we flush per bulksize or after a certain time has elapsed?

Comment (vwidjaya) [reply flow/elastic.go]: Both are always applied, and the one met first will be triggered. If neither of them are specified, the default values will be used anyways so I decided not to separate them into if-else statements. Should I

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FROM ykode/golang-builder as builder
 
-WORKDIR /go/src/github.com/BaritoLog/barito-flow
+WORKDIR /go/src/barito-flow
```
[the remark this answers is not in the document]
Comment (vwidjaya) [reply docker/Dockerfile]: does using "module github.com/BaritoLog/barito-flow" sound right for package naming?

File: flow/elastic.go (hunk 2 above)
Comment (giosakti) [reply flow/elastic.go]: okay cool, I just need to understand the flow

File: docker/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (giosakti) [reply docker/Dockerfile]: sorry, I meant for this Dockerfile only. not to change all other files
is it still working for local development even if we add `github.com/BaritoLog` prefix?

**Pull request merged**